c/meterpreter/source/extensions/priv/service.c

```diff
@@ -172,3 +172,47 @@ DWORD service_destroy( char * cpName )
 
 	return dwResult;
 }
+
+/*
+ * Query service state.
+ */
+DWORD query_service_status( char* cpName, DWORD* dwState )
+{
+	DWORD dwResult  = ERROR_SUCCESS;
+	HANDLE hManager = NULL;
+	HANDLE hService = NULL;
+	DWORD dwBytes;
+	SERVICE_STATUS_PROCESS procInfo;
+
+	do
+	{
+		if( !cpName )
+			BREAK_WITH_ERROR( "[SERVICE] query_service_status. cpName is NULL", ERROR_INVALID_HANDLE );
+
+		hManager = OpenSCManagerA( NULL, NULL, SC_MANAGER_ALL_ACCESS );
```

Comment:
SC_MANAGER_ALL_ACCESS is excessive when you just want to query the service and may lead to denials as you can't gain all the privileges requested. Always request the minimum privileges needed to do your job to prevent this and other related issues. Perhaps GENERIC_READ is better suited here given we are just querying?

Reply:
`SC_MANAGER_CONNECT` is enough for a simple query in our case 👍